lms/views/authentication.py

```diff
@@ -14,7 +14,7 @@ def __init__(self, request):
         self.logged_in = request.authenticated_userid
 
     @view_config(route_name="login", renderer="templates/login.html.jinja2")
-    @forbidden_view_config(renderer="templates/login.html.jinja2")
+    @forbidden_view_config(renderer="templates/login.html.jinja2", route_name="reports")
```

Comment:
I assume the normal `view_config` here is so that this view can be accessed both directly and as a result of a failed lookup for an authenticated view?

Reply:
Yes. If you visit `/reports` while not logged in you'll see that the login form is rendered at the `/reports` URL. I'm not sure that's the right thing to do (rather than redirecting to `/login`) but it's preexisting behavior.